src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/StringParsingHelpers.Addresses.cs

```diff
@@ -12,6 +12,11 @@ internal static partial class StringParsingHelpers
         // and seperates the information about by each interface.
         internal static List<GatewayIPAddressInformation> ParseGatewayAddressesFromRouteFile(string filePath, string interfaceName)
         {
+            if (!File.Exists(filePath))
+            {
+                throw ExceptionHelper.CreateForInformationUnavailable();
+            }
```

Comment:
Do we need to worry about a race condition with the file going away after this check, or is this just a case of the file existing or not on a given platform, and if it's there it's always going to be there?